source/GameInterface/Services/CraftingService/Patches/CraftingLifetimePatches.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using GameInterface.Policies;
+using GameInterface.Services.CraftingService.Messages;
+using GameInterface.Services.ObjectManager;
+using HarmonyLib;
+using Serilog;
+using System;
+using TaleWorlds.CampaignSystem.GameState;
+using TaleWorlds.CampaignSystem.MapEvents;
+using TaleWorlds.Core;
+using TaleWorlds.Localization;
+
+namespace GameInterface.Services.CraftingService.Patches
+{
+    [HarmonyPatch(typeof(Crafting))]
+    public class CraftingLifetimePatches
+    {
+        private static readonly ILogger Logger = LogManager.GetLogger<CraftingLifetimePatches>();
+
+        [HarmonyPatch(typeof(Crafting), MethodType.Constructor, typeof(CraftingTemplate), typeof(BasicCultureObject), typeof(TextObject))]
+        [HarmonyPrefix]
+        static bool CreateCraftingPrefix(ref Crafting __instance, CraftingTemplate craftingTemplate, BasicCultureObject culture, TextObject name)
+        {
+            // Call original if we call this function
+            if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+            if (ModInformation.IsClient)
+            {
+                Logger.Error("Client created unmanaged {name}\n"
+                    + "Callstack: {callstack}", typeof(Crafting), Environment.StackTrace);
+
+                return true;
+            }
+
+            if (ContainerProvider.TryResolve<IObjectManager>(out var objectManager))
+            {
+                objectManager.AddNewObject(__instance, out var newId);
+
+                var data = new CraftingCreatedData(newId, craftingTemplate.StringId, culture.StringId, name.Value);
+                var message = new CraftingCreated(data);
+
+                MessageBroker.Instance.Publish(null, message);
+            }
```

Comment:
Can you pass up the instances to the lifetime handler and resolve the ids in there?